toolkit/tools/graphpkgfetcher/graphpkgfetcher.go

```diff
@@ -110,13 +119,85 @@ func fetchPackages() (err error) {
 		return
 	}
 
+	// Optional delta build cache hydration
+	if *tryDownloadDeltaRPMs {
+		logger.Log.Info("Attempting to download delta RPMs for build nodes")
+		err = downloadDeltaNodes(dependencyGraph, cloner)
+		if err != nil {
+			err = fmt.Errorf("failed to download delta RPMs: %w", err)
+			return
+		}
+		// Update the package graph with the paths to the delta RPMs we downloaded
+		err = pkggraph.WriteDOTGraphFile(dependencyGraph, *outputGraph)
+		if err != nil {
+			err = fmt.Errorf("failed to write cache graph to file: %w", err)
+			return
+		}
+	}
+
 	// If we grabbed any RPMs, we need to convert them into a local repo
-	if hasUnresolvedNodes {
+	if *tryDownloadDeltaRPMs || hasUnresolvedNodes {
+		logger.Log.Info("Configuring downloaded RPMs as a local repository")
 		err = cloner.ConvertDownloadedPackagesIntoRepo()
 		if err != nil {
 			err = fmt.Errorf("failed to convert downloaded RPMs into a repo:\n%w", err)
 			return err
 		}
+
+		if strings.TrimSpace(*outputSummaryFile) != "" {
+			err = repoutils.SaveClonedRepoContents(cloner, *outputSummaryFile)
+			if err != nil {
+				logger.Log.Errorf("Failed to save cloned repo contents.")
```

Comment:
>logger.Log.Errorf("Failed to save cloned repo contents.") [](http://example.com/codeflow?start=4&length=57)

Since we're doing this elsewhere, why not:

`err = fmt.Errorf("failed to save cloned repo contents:\n%w", err)`